nominal/experimental/stream_v2/_write_stream.py

```diff
@@ -58,17 +63,38 @@ def create(
             max_queue_size=batch_queue_maxsize,
         )
         batch_serialize_thread = spawn_batch_serialize_thread(
-            write_pool, clients, serializer, nominal_data_source_rid, batch_queue
+            write_pool, clients, serializer, nominal_data_source_rid, batch_queue, item_queue, track_metrics
         )
+
+        def add_metric_impl(channel_name: str, timestamp: int, value: float) -> None:
+            item_queue.put(
+                BatchItem(
+                    channel_name=channel_name,
+                    timestamp=timestamp,
+                    value=value,
+                )
+            )
+
+        def add_metric_noop(channel_name: str, timestamp: int, value: float) -> None:
+            pass
+
+        add_metric_fn = add_metric_impl if track_metrics else add_metric_noop
+
         return cls(
             _write_pool=write_pool,
             _item_queue=item_queue,
             _batch_thread=batch_thread,
             _batch_serialize_thread=batch_serialize_thread,
             _serializer=serializer,
+            _track_metrics=track_metrics,
             _clients=clients,
+            _add_metric=add_metric_fn,
         )
 
+    def add_metric(self, channel_name: str, timestamp: int, value: float) -> None:
```

Comment:
`int` or `IntegralNanosecondsUTC`?

Reply:
I think I wouldn't expose this method in the public API and just keep the internal `_` method